.github/workflows/publish-to-pypi.yml

```diff
@@ -2,21 +2,31 @@ name: Publish releases to PyPI
 
 on:
   release:
-    types: [published, prereleased]
+    types: [published]
 
 jobs:
   build-and-publish:
     name: Builds and publishes releases to PyPI
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3.5.3
+      - name: Get tag
+        id: vars
+        run: echo "tag=${GITHUB_REF#refs/*/}" >> $GITHUB_OUTPUT
       - name: Set up Python 3.10
         uses: actions/setup-python@v4.6.1
         with:
           python-version: "3.10"
       - name: Install build
         run: >-
           pip install build
+      - name: Validate tag matches Python project version
+        shell: python
+        run: |-
+          import tomli
+          with open("pyproject.toml", "rb") as f:
+            pyproject = tomli.load(f)
+          assert pyproject["project"]["version"] == "${{ steps.vars.outputs.tag }}"
```

Comment:
Alternatively (and IMHO better solution), is to not set it and update it on release automatically.

Reply:
Hm, so that would involve reading pyproject.toml here, setting the version and writing it back out?

Reply:
I've implemented that in a new PR, see #326.